### ISSUE TITLE
divergent transitions - change adapt_delta

### ISSUE THREAD

**Issue by lmguzman** [gbradburd/conStruct#17]
Hi everyone, 

First, thanks for such a great package with great documentation. I appreciate it.

I just had one 'hopefully' quick question. I am able to successfully run conStruct with my data, but I get the following warning message:

Warning messages:
1: There were 193 divergent transitions after warmup. Increasing adapt_delta above 0.8 may help. See
http://mc-stan.org/misc/warnings.html#divergent-transitions-after-warmup 
2: Examine the pairs() plot to diagnose sampling problems

I checked the website the warning message recommends, and it says that adapt_delta should be a parameter in any of the stan functions. 

I just wanted to check with you, if there is a quick way of doing this from your functions, or should I clone the package and try to change it internally?

Thank you!

**Comment by petrelharp** [comment]
There doesn't seem to be an easy way to do this currently; but I've proposed a change that'd allow it: #18

**Comment by lmguzman** [comment]
Thank you!

**Comment by gbradburd** [comment]
I've merged Peter's fix into master, so if you re-install the package from the git repo (with, e.g., `install_github`), you should be able to adjust the `adapt_delta` parameter.  Note that the adaptation parameters are passed to stan via a named `control` list, so if you want to change the `adapt_delta` parameter, you have to do something like:

```
conStruct(spatial = TRUE,
	 K = 2,
	 freqs = myFreqs,
	 geoDist = myDist,
	 coords = myCoords,
	 prefix = myPrefix,
	 control = setNames(list(0.9),"adapt_delta"))
```

Let me know if you still have any questions about this, and, if not, feel free to close the issue.

**Comment by lmguzman** [comment]
Thank you! Now it works!

**Comment by oscaredd** [comment]
Hi there!

I would like to get your advice for a problem I have with the mixing of chains in my analysis. My dataset contains ~600 SNPs (those have shown nice results with traditional clustering methods) with 64 "samples", with allele frequencies calculated from one or several individuals.
For both discrete and spatial methods in conStruct, I keep getting both R-hat (it has reached ~4.5 in some cases) and the "divergent transitions after warmup" warnings. So far I've tried increasing iterations from 10K to 100K, increasing `adapt _delta` up to 0.99, run 2 or 4 chains. In all cases I've ran 2 independent analyses for each K (1-8). To this moment, I can only think of these possibilities: 1) need more SNPs (I could go up to ~43K SNPs, but then some of those will probably by linked), and 2) need to reduce complexity by reducing the number of samples.

Thanks in advance for your help!
